Microsoft.Toolkit.Services/Services/LinkedIn/LinkedInDataProvider.cs

```diff
@@ -81,7 +81,8 @@ public LinkedInDataProvider(LinkedInOAuthTokens tokens, LinkedInPermissions requ
                 throw new ArgumentException("Missing callback uri");
             }
 
-            if (!Enum.IsDefined(typeof(LinkedInPermissions), requiredPermissions))
+            // Check if its a valid combination of LinkedInPermissions
+            if ((~(int)LinkedInPermissionsHelpers.AllPermissions & (int)requiredPermissions) == 0)
```

Comment:
Wait, my original sample had the _valid_ path inside the `if` branch.
If you want that same conditional path to throw you need to invert the check:

```suggestion
            if ((~(int)LinkedInPermissionsHelpers.AllPermissions & (int)requiredPermissions) != 0)
```

ie. a flags combination is invalid if there's any set bit outside the bitmask with all flags set (so `!= 0`) 😊

Reply:
Ops 😶